src/test/java/org/eclipse/uprotocol/rpc/RpcTest.java

```diff
@@ -64,8 +73,10 @@ public CompletableFuture<UPayload> invokeMethod(UUri topic, UPayload payload, UA
         public CompletableFuture<UPayload> invokeMethod(UUri topic, UPayload payload, UAttributes attributes) {
             Status status = Status.newBuilder().setCode(Code.INVALID_ARGUMENT_VALUE).setMessage("boom").build();
             Any any = Any.pack(status);
-            UPayload data = new UPayload(any.toByteArray(), UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF);
-
+            UPayload data = UPayload.newBuilder()
```

Comment:
personally I prefer working with the constructors and not verbose builders if they are not really needed. 
I do understand why you wanted to define the payload as a protobuf object
But then again - at the end of the day, uTransport gets a byte[], so are we not creating a protobuf object whose job is transport and then just translating it into the **real** transport object which is the byte[]
what value is achieved by the payload being a protobuf object?

Reply:
UPayload is to wrap the data + metadata and keep it separate from the message attributes (priority, addressing, timeout, etc...). Yes you are right for the data but the metadata will most likely be mapped somehow into the local transport (HTTP header, zenoh header, etc...). 
the benefit of the UPayload being in proto is the consistently for all data types (and the UPayloadFormat as well).